topics/ui/controls/checkbox.topic

```diff
@@ -0,0 +1,207 @@
+                column.setCellEditor(JBTable.createBooleanEditor());
+                column.setCellRenderer(new BooleanTableCellRender());
+            </code-block>
+                </chapter>
+        </chapter>
+            <chapter title="Three-state checkbox" id="three-state-checkbox">
+                <p>In a group of options, use the parent checkbox to show the status of its children:</p>
+                <img src="checkbox_three_state.png"
+                     alt="Different states for a parent checkbox: checked, indeterminate, and unchecked" width="706"/>
+                <p>The parent checkbox in checked, indeterminate and unchecked states.</p>
+                    <chapter title="Implementation" collapsible="true">
+                    The three-state checkbox is represented by the <a
+                            href="%gh-ic%/platform/util/ui/src/com/intellij/util/ui/ThreeStateCheckBox.java"><code>ThreeStateCheckBox</code></a>
+                    class which represents its state with the <code>ThreeStateCheckBox.State</code> enum containing <code>SELECTED,
+                        NOT_SELECTED, DONT_CARE</code> states.</chapter>
+                <p>When the user clicks an indeterminate checkbox for the first time, the whole group becomes checked. The
+                    second click unchecks the whole group.</p>
+                <p>An indeterminate checkbox can also show the download status. An example with a remote repository:</p>
+                <img src="checkbox_download_status.png" alt="Indeterminate checkboxes showing download status" width="706"/>
+                <p>
+                    Repositories &quot;tools-base&quot; and &quot;contrib&quot; are being loaded. When loading is
+                    finished, the indeterminate checkbox will be replaced with the checked checkbox if there are
+                    commits, or an unchecked checkbox if there are no commits.
+                </p>
+                <chapter title="Implementation" collapsible="true">
+                    <p>In a table, the three-state checkbox is represented by <a href="%gh-ic%/platform/lang-impl/src/com/intellij/profile/codeInspection/ui/table/ThreeStateCheckBoxRenderer.java"><code>ThreeStateCheckBoxRenderer</code></a>
+                        that provides both <code>TableCellRenderer</code> and <code>TableEditor</code>.
+                        It accepts <code>Boolean</code> type in the column being supplied by the <code>TableModel</code> and
+                        becomes <code>DONT_CARE</code> when the value in the cell is null.
+                        Otherwise, it becomes <code>SELECTED</code> for <code>Boolean.TRUE</code>, and <code>NOT_SELECTED</code>
+                        for <code>Boolean.FALSE</code>.</p>
+                </chapter>
+            </chapter>
+        <chapter title="Writing guidelines" id="writing-guidelines">
+                <list>
+                    <li>Use sentence-style capitalization:
```

Comment:
Bullet points seem a bit noisy. Illustrations already work as separators between rules, possibly could be ok without bullet points?

Reply:
I agree, I removed them